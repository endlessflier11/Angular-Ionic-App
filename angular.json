{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "defaultProject": "app",
  "newProjectRoot": "projects",
  "projects": {
    "app": {
      "root": "",
      "sourceRoot": "src",
      "projectType": "application",
      "prefix": "csaa",
      "schematics": {},
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "www",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.app.json",
            "assets": [
              {
                "glob": "**/*",
                "input": "src/assets",
                "output": "assets"
              },
              {
                "glob": "**/*",
                "input": "src/app/micro-apps/csaa-mobile/assets",
                "output": "assets/csaa-mobile"
              },
              {
                "glob": "**/*.svg",
                "input": "node_modules/ionicons/dist/ionicons/svg",
                "output": "./svg"
              }
            ],
            "styles": [
              {
                "input": "src/theme/variables.scss"
              },
              {
                "input": "src/global.scss"
              }
            ],
            "scripts": [],
            "allowedCommonJsDependencies": [
              "us-bank-account-validator",
              "card-validator",
              "crypto-js"
            ],
            "buildOptimizer": false,
            "sourceMap": true,
            "optimization": false
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "namedChunks": false,
              "aot": true,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                }
              ]
            },
            "ds": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.ds.ts"
                }
              ]
            },
            "development": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.ds.ts"
                }
              ]
            },
            "ci": {
              "progress": false
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "app:build",
            "proxyConfig": "proxy.conf.js"
          },
          "configurations": {
            "production": {
              "browserTarget": "app:build:production",
              "proxyConfig": "proxy.prod.conf.json"
            },
            "development": {
              "browserTarget": "app:build:development",
              "proxyConfig": "proxy.conf.json"
            },
            "ds": {
              "browserTarget": "app:build:ds",
              "proxyConfig": "proxy.conf.json"
            },
            "ci": {
              "progress": false
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "app:build"
          }
        },
        "test": {
          "builder": "@angular-builders/jest:run",
          "options": {
            "tsConfig": "tsconfig.spec.json"
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": ["src/**/*.ts", "src/**/*.html"]
          }
        },
        "e2e": {
          "builder": "@angular-devkit/build-angular:protractor",
          "options": {
            "protractorConfig": "e2e/protractor.conf.js",
            "devServerTarget": "app:serve"
          },
          "configurations": {
            "production": {
              "devServerTarget": "app:serve:production"
            },
            "ci": {
              "devServerTarget": "app:serve:ci"
            }
          }
        },
        "ionic-cordova-build": {
          "builder": "@ionic/angular-toolkit:cordova-build",
          "options": {
            "browserTarget": "app:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "app:build:production"
            }
          }
        },
        "ionic-cordova-serve": {
          "builder": "@ionic/angular-toolkit:cordova-serve",
          "options": {
            "cordovaBuildTarget": "app:ionic-cordova-build",
            "devServerTarget": "app:serve"
          },
          "configurations": {
            "production": {
              "cordovaBuildTarget": "app:ionic-cordova-build:production",
              "devServerTarget": "app:serve:production"
            }
          }
        }
      }
    }
  },
  "cli": {
    "defaultCollection": "@angular-eslint/schematics",
    "analytics": false
  },
  "schematics": {
    "@ionic/angular-toolkit:component": {
      "styleext": "scss"
    },
    "@ionic/angular-toolkit:page": {
      "styleext": "scss"
    }
  }
}
