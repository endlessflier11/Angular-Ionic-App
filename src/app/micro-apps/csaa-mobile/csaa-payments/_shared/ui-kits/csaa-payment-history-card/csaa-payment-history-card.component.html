<csaa-skeleton-card *ngIf="loading"></csaa-skeleton-card>
<ion-card *ngIf="!loading" class="csaa-card">
  <ion-card-header class="policy-number-container" *ngIf="policy?.number">
    <ion-item [lines]="headerLine" class="ion-text-wrap ion-no-padding csaa-policy-item">
      <ion-label>
        <h3>{{ policy?.number }}</h3>
        <p class="subtitle">
          <csaa-policy-subtitles [policySubtitle]="policy.subtitle"></csaa-policy-subtitles>
        </p>
      </ion-label>
      <csaa-theme-icon slot="end" [type]="policy.type" [size]="40"></csaa-theme-icon>
    </ion-item>
  </ion-card-header>
  <ion-card-content class="csaa-item-list">
    <ion-list>
      <ion-item
        *ngFor="let payment of paymentHistory"
        [class.failed]="getPaymentStatus(payment.status) === status.FAILED"
      >
        <ion-label>
          <p class="payment-date ion-text-left">
            {{
              payment.transactionDate.split('-').slice(0, 3).join('-') | date: 'mediumDate':timezone
            }}
          </p>
          <p class="status ion-text-left">
            {{ getPaymentStatus(payment.status) }}
          </p>
          <p class="status ion-text-left">
            {{ getPaymentMethod(payment.paymentMethod) }}
          </p>
        </ion-label>
        <span slot="end" class="amount ion-text-right">
          {{ payment.amount | currency: 'USD':'symbol':'1.2-2' }}
        </span>
      </ion-item>
    </ion-list>
  </ion-card-content>
</ion-card>
