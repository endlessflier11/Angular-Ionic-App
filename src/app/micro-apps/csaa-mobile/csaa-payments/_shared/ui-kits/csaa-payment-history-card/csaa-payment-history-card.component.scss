@import '../../../../_scss/variables';

ion-card {
  ion-card-header {
    padding-bottom: 0;
    padding-right: 15px;
    padding-left: 15px;
  }
  ion-card-content {
    padding-inline-start: 15px;
    padding-inline-end: 15px;
  }

  ion-item {
    &.failed {
      .amount,
      .status,
      &:before {
        color: $csaa-emergency-color;
      }
    }
  }
}
