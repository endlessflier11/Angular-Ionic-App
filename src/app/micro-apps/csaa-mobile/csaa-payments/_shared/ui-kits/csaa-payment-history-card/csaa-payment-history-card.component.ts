import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
import {
  Policy,
  PaymentHistory,
  PaymentHistoryEntry,
  PaymentStatus,
  PaymentMethod,
} from '../../../../_core/interfaces';

@Component({
  selector: 'csaa-payment-history-card',
  templateUrl: 'csaa-payment-history-card.component.html',
  styleUrls: ['csaa-payment-history-card.component.scss'],
})
export class CsaaPaymentHistoryCardComponent implements OnChanges {
  @Input() policy: Policy;
  @Input() loading = false;
  @Input() headerLine = 'full';
  @Input() history: PaymentHistory;
  @Input() timezone: string = undefined;

  paymentHistory: PaymentHistoryEntry[];
  status = PaymentStatus;

  ngOnChanges(changes: SimpleChanges) {
    const historyIn =
      changes.history &&
      changes.history.currentValue &&
      Object.keys(changes.history.currentValue) &&
      Object.keys(changes.history.currentValue).length > 0;
    if (historyIn && this.policy && this.policy.number) {
      this.paymentHistory = changes.history.currentValue[this.policy.number].slice(0, 4);
    }
  }

  getPaymentStatus(paymentStatus): PaymentStatus {
    const status = paymentStatus ? paymentStatus.toLowerCase() : '';
    switch (status) {
      case 'issued':
        return PaymentStatus.PAID;
      case 'pending':
        return PaymentStatus.PENDING;
      default:
        return PaymentStatus.FAILED;
    }
  }

  getPaymentMethod(paymentMethod): PaymentMethod {
    const method = paymentMethod ? paymentMethod.toLowerCase() : '';
    switch (method) {
      case 'cash':
        return PaymentMethod.CASH;
      case 'cheque':
        return PaymentMethod.CHECK;
      case 'creditcard':
      case 'pcicreditcard':
        return PaymentMethod.CREDIT_CARD;
      case 'debitcard':
      case 'pcidebitcard':
        return PaymentMethod.DEBIT_CARD;
      case 'eft':
        return PaymentMethod.EFT;
      default:
        return PaymentMethod.EMPTY;
    }
  }
}
