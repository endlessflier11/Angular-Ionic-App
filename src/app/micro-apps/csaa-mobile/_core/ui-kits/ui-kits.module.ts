import { NgModule } from '@angular/core';
import { CommonModule, DatePipe } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { IonicModule } from '@ionic/angular';
import { ThemeIconComponent } from './theme-icon/theme-icon.component';
import { WorkingHoursComponent } from './working-hours/working-hours.component';
import { MyPolicyCardComponent } from './my-policy-card/my-policy-card.component';
import { SkeletonCardComponent } from './skeleton-card/skeleton-card.component';
import { BackButtonComponent } from './back-button/back-button.component';
import { PhonePipe } from './pipes/phone-number.pipe';
import { CsaaAutoIdCardComponent } from './csaa-auto-id-card/csaa-auto-id-card.component';
import { CsaaLoadingComponent } from './csaa-loading/csaa-loading.component';
import { ContentRendererComponent } from './content-renderer/content-renderer.component';
import { NationalCustomAlertModalComponent } from './national-custom-alert/national-custom-alert-modal';
import { SafeDataPipe } from './pipes/safe-data/safe-data.pipe';
import { DocumentActionModalComponent } from './document-action-modal/document-action-modal.component';
import { CsaaPolicySubtitlesComponent } from './csaa-policy-subtitles/csaa-policy-subtitles.component';
import { PaperlessTermsComponent } from './paperless-terms/paperless-terms.component';
import { PaperlessTermsModalComponent } from './paperless-terms-modal/paperless-terms-modal.component';
import { AppleWalletButtonComponent } from './apple-wallet-button/apple-wallet-button.component';

const UI_COMPONENTS = [
  ThemeIconComponent,
  WorkingHoursComponent,
  MyPolicyCardComponent,
  SkeletonCardComponent,
  BackButtonComponent,
  CsaaAutoIdCardComponent,
  CsaaLoadingComponent,
  ContentRendererComponent,
  NationalCustomAlertModalComponent,
  DocumentActionModalComponent,
  CsaaPolicySubtitlesComponent,
  PaperlessTermsComponent,
  PaperlessTermsModalComponent,
  AppleWalletButtonComponent,
];

const UI_PIPES = [PhonePipe, SafeDataPipe];

@NgModule({
  imports: [CommonModule, ReactiveFormsModule, IonicModule],
  declarations: [...UI_COMPONENTS, ...UI_PIPES],
  exports: [...UI_COMPONENTS, ...UI_PIPES],
  providers: [DatePipe],
})
export class UiKitsModule {}
