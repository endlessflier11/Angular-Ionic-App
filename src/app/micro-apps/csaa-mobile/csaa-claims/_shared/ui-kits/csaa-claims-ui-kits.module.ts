import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { AgentCardComponent } from './agent-card/agent-card.component';
import { ClaimsHistoryCardComponent } from './claims-history-card/claims-history-card.component';
import { ClaimsInfoCardComponent } from './claims-info-card/claims-info-card.component';
import { WhatToDoCardComponent } from './what-to-do-card/what-to-do-card.component';

const SHARED_COMPONENTS = [
  AgentCardComponent,
  ClaimsHistoryCardComponent,
  ClaimsInfoCardComponent,
  WhatToDoCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaClaimsUiKitsModule {}
