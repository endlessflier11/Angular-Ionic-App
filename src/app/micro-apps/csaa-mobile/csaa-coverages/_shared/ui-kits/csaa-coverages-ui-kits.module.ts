import { NgModule } from '@angular/core';
import { CsaaPupCoveragesCardComponent } from './csaa-pup-coverages-card/csaa-pup-coverages-card.component';
import { CsaaGeneralCoveragesCardComponent } from './csaa-general-coverages-card/csaa-general-coverages-card.component';
import { CsaaDriversCoveragesCardComponent } from './csaa-drivers-coverages-card/csaa-drivers-coverages-card.component';
import { CsaaHomeCoveragesCardComponent } from './csaa-home-coverages-card/csaa-home-coverages-card.component';
import { CsaaVehiclesCoveragesCardComponent } from './csaa-vehicles-coverages-card/csaa-vehicles-coverages-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaCoverageCardItemComponent } from './csaa-coverage-card-item/csaa-coverage-card-item.component';
import {
  CsaaCoveragesVehicleCoveredCardComponent
} from './csaa-vehicle-coverages-covered-card/csaa-coverages-vehicle-covered-card.component';
import {
  CsaaCoveragesVehicleNotCoveredCardComponent
} from './csaa-vehicle-coverages-not-covered-card/csaa-coverages-vehicle-not-covered-card.component';
import { CsaaDriversCoveragesContentV1Component } from './csaa-drivers-coverages-content-v1/csaa-drivers-coverages-content-v1.component';
import { CsaaDriversCoveragesContentV2Component } from './csaa-drivers-coverages-content-v2/csaa-drivers-coverages-content-v2.component';

const SHARED_COMPONENTS = [
  CsaaCoverageCardItemComponent,
  CsaaPupCoveragesCardComponent,
  CsaaGeneralCoveragesCardComponent,
  CsaaDriversCoveragesCardComponent,
  CsaaDriversCoveragesContentV1Component,
  CsaaHomeCoveragesCardComponent,
  CsaaVehiclesCoveragesCardComponent,
  CsaaCoveragesVehicleCoveredCardComponent,
  CsaaCoveragesVehicleNotCoveredCardComponent,
  CsaaDriversCoveragesContentV2Component,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaCoveragesUiKitsModule {}
