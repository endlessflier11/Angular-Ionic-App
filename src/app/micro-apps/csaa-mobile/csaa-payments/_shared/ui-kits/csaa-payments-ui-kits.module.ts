import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaPaymentHistoryCardComponent } from './csaa-payment-history-card/csaa-payment-history-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentHistorySelectCardComponent } from './csaa-payment-history-select-card/csaa-payment-history-select-card.component';
import { CsaaPaymentSucceededCardComponent } from './csaa-payment-succeeded-card/csaa-payment-succeeded-card.component';
import { CsaaMakePaymentCardComponent } from './csaa-make-payment-card/csaa-make-payment-card.component';
import { CsaaPayAllPoliciesCardComponent } from './csaa-pay-all-policies-card/csaa-pay-all-policies-card.component';
import { PaymentTermsConditionsComponent } from './payment-terms-conditions/payment-terms-conditions.component';
import { AutopayTermsConditionsComponent } from './autopay-terms-conditions/autopay-terms-conditions.component';
import { CsaaBillsCardComponent } from './csaa-bills-card/csaa-bills-card.component';
import { AutopayDiscountCardComponent } from './autopay-discount-card/autopay-discount-card.component';

const SHARED_COMPONENTS = [
  CsaaPaymentHistoryCardComponent,
  CsaaPaymentHistorySelectCardComponent,
  CsaaMakePaymentCardComponent,
  CsaaPaymentSucceededCardComponent,
  CsaaPayAllPoliciesCardComponent,
  PaymentTermsConditionsComponent,
  AutopayTermsConditionsComponent,
  CsaaBillsCardComponent,
  AutopayDiscountCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule, IonicModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaPaymentsUiKitsModule {}
