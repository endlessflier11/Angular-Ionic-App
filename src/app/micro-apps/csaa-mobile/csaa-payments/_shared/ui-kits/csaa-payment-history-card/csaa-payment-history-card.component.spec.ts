import { CommonModule } from '@angular/common';
import { NO_ERRORS_SCHEMA, SimpleChange } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { IonicModule } from '@ionic/angular';
import { By } from '../../../../../../../testing';
import { PageTestingModule } from '../../../../../../../testing/misc/page-testing.module';
import { PaymentHistory } from '../../../../_core/interfaces/payment.interface';
import { Policy, PolicyStatus } from '../../../../_core/interfaces/policy.interface';
import { UiKitsModule } from '../../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentHistoryCardComponent } from './csaa-payment-history-card.component';

describe('CsaaPaymentHistoryCardComponent', () => {
  let component: CsaaPaymentHistoryCardComponent;
  let fixture: ComponentFixture<CsaaPaymentHistoryCardComponent>;
  const mockPolicyNumber = '12345';

  beforeEach(async () => {
    TestBed.configureTestingModule({
      declarations: [CsaaPaymentHistoryCardComponent],
      imports: [
        CommonModule,
        IonicModule,
        UiKitsModule,
        PageTestingModule.withConfig({
          providesStorage: true,
          providesAnalytics: true,
          providesConfig: true,
        }),
      ],
      schemas: [NO_ERRORS_SCHEMA],
    });

    const mockHistory: PaymentHistory = {
      [mockPolicyNumber]: [
        {
          transactionDate: '2018-08-10',
          amount: '148.1',
          status: 'paid',
          paymentMethod: 'creditcard',
          policyEffectiveDate: '2018-08-10',
          policyExpirationDate: '2018-08-10',
          referenceNumber: '1',
          subType: 'string',
          userID: '1',
        },
        {
          transactionDate: '2018-08-10',
          amount: '148.1',
          status: 'pending',
          paymentMethod: 'creditcard',
          policyEffectiveDate: '2018-08-10',
          policyExpirationDate: '2018-08-10',
          referenceNumber: '1',
          subType: 'string',
          userID: '1',
        },
        {
          transactionDate: '2018-08-10',
          amount: '148.1',
          status: 'failed',
          paymentMethod: 'creditcard',
          policyEffectiveDate: '2018-08-10',
          policyExpirationDate: '2018-08-10',
          referenceNumber: '1',
          subType: 'string',
          userID: '1',
        },
        {
          transactionDate: '2018-08-10',
          amount: '148.1',
          status: 'paid',
          paymentMethod: 'creditcard',
          policyEffectiveDate: '2018-08-10',
          policyExpirationDate: '2018-08-10',
          referenceNumber: '1',
          subType: 'string',
          userID: '1',
        },
        {
          transactionDate: '2018-08-10',
          amount: '148.1',
          status: 'paid',
          paymentMethod: 'creditcard',
          policyEffectiveDate: '2018-08-10',
          policyExpirationDate: '2018-08-10',
          referenceNumber: '1',
          subType: 'string',
          userID: '1',
        },
        {
          transactionDate: '2018-08-10',
          amount: '148.1',
          status: 'paid',
          paymentMethod: 'creditcard',
          policyEffectiveDate: '2018-08-10',
          policyExpirationDate: '2018-08-10',
          referenceNumber: '1',
          subType: 'string',
          userID: '1',
        },
      ],
    };

    const mockPolicy: Policy = {
      type: 2,
      id: '1',
      subtitle: '2018 Porsche 911',
      number: mockPolicyNumber,
      status: PolicyStatus.ACTIVE,
      riskState: 'AZ',
      productCode: '1',
      policyPrefix: 'AU_SS',
    };

    fixture = TestBed.createComponent(CsaaPaymentHistoryCardComponent);
    component = fixture.componentInstance;
    component.history = mockHistory;
    component.policy = mockPolicy;
    component.ngOnChanges({ history: new SimpleChange(null, mockHistory, false) });
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot when loading and no payments', () => {
    component.loading = true;
    expect(fixture).toMatchSnapshot();
  });

  it('should only show the last 4 payments', () => {
    expect(fixture.debugElement.queryAll(By.cssAndText('ion-item', 'Credit Card')).length).toBe(4);
  });
});
