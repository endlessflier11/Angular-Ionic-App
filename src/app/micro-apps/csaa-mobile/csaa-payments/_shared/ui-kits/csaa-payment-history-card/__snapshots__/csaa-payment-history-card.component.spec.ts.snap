// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaPaymentHistoryCardComponent should match snapshot 1`] = `
<csaa-payment-history-card
  __ngContext__={[Function LRootView]}
  headerLine={[Function String]}
  history={[Function Object]}
  loading="false"
  paymentHistory={[Function Array]}
  policy={[Function Object]}
  status={[Function Object]}
  timezone="undefined"
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header
      class="policy-number-container"
    >
      <ion-item
        class="ion-text-wrap ion-no-padding csaa-policy-item"
      >
        <ion-label>
          <h3>
            12345
          </h3>
          <p
            class="subtitle"
          >
            <csaa-policy-subtitles>
              <span
                class="subtitle"
              >
                2018 Porsche 911
              </span>
            </csaa-policy-subtitles>
          </p>
        </ion-label>
        <csaa-theme-icon
          slot="end"
        >
          <img
            src="/assets/csaa-mobile/vectors/icn-Auto.svg"
            style="width: 40px; height: 40px;"
          />
        </csaa-theme-icon>
      </ion-item>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item
          class="failed"
        >
          <ion-label>
            <p
              class="payment-date ion-text-left"
            >
               Aug 10, 2018 
            </p>
            <p
              class="status ion-text-left"
            >
               Failed 
            </p>
            <p
              class="status ion-text-left"
            >
               Credit Card 
            </p>
          </ion-label>
          <span
            class="amount ion-text-right"
            slot="end"
          >
             $148.10 
          </span>
        </ion-item>
        <ion-item>
          <ion-label>
            <p
              class="payment-date ion-text-left"
            >
               Aug 10, 2018 
            </p>
            <p
              class="status ion-text-left"
            >
               Pending 
            </p>
            <p
              class="status ion-text-left"
            >
               Credit Card 
            </p>
          </ion-label>
          <span
            class="amount ion-text-right"
            slot="end"
          >
             $148.10 
          </span>
        </ion-item>
        <ion-item
          class="failed"
        >
          <ion-label>
            <p
              class="payment-date ion-text-left"
            >
               Aug 10, 2018 
            </p>
            <p
              class="status ion-text-left"
            >
               Failed 
            </p>
            <p
              class="status ion-text-left"
            >
               Credit Card 
            </p>
          </ion-label>
          <span
            class="amount ion-text-right"
            slot="end"
          >
             $148.10 
          </span>
        </ion-item>
        <ion-item
          class="failed"
        >
          <ion-label>
            <p
              class="payment-date ion-text-left"
            >
               Aug 10, 2018 
            </p>
            <p
              class="status ion-text-left"
            >
               Failed 
            </p>
            <p
              class="status ion-text-left"
            >
               Credit Card 
            </p>
          </ion-label>
          <span
            class="amount ion-text-right"
            slot="end"
          >
             $148.10 
          </span>
        </ion-item>
      </ion-list>
    </ion-card-content>
  </ion-card>
</csaa-payment-history-card>
`;

exports[`CsaaPaymentHistoryCardComponent should match snapshot when loading and no payments 1`] = `
<csaa-payment-history-card
  __ngContext__={[Function LRootView]}
  headerLine={[Function String]}
  history={[Function Object]}
  loading={[Function Boolean]}
  paymentHistory={[Function Array]}
  policy={[Function Object]}
  status={[Function Object]}
  timezone="undefined"
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header
      class="policy-number-container"
    >
      <ion-item
        class="ion-text-wrap ion-no-padding csaa-policy-item"
      >
        <ion-label>
          <h3>
            12345
          </h3>
          <p
            class="subtitle"
          >
            <csaa-policy-subtitles>
              <span
                class="subtitle"
              >
                2018 Porsche 911
              </span>
            </csaa-policy-subtitles>
          </p>
        </ion-label>
        <csaa-theme-icon
          slot="end"
        >
          <img
            src="/assets/csaa-mobile/vectors/icn-Auto.svg"
            style="width: 40px; height: 40px;"
          />
        </csaa-theme-icon>
      </ion-item>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item
          class="failed"
        >
          <ion-label>
            <p
              class="payment-date ion-text-left"
            >
               Aug 10, 2018 
            </p>
            <p
              class="status ion-text-left"
            >
               Failed 
            </p>
            <p
              class="status ion-text-left"
            >
               Credit Card 
            </p>
          </ion-label>
          <span
            class="amount ion-text-right"
            slot="end"
          >
             $148.10 
          </span>
        </ion-item>
        <ion-item>
          <ion-label>
            <p
              class="payment-date ion-text-left"
            >
               Aug 10, 2018 
            </p>
            <p
              class="status ion-text-left"
            >
               Pending 
            </p>
            <p
              class="status ion-text-left"
            >
               Credit Card 
            </p>
          </ion-label>
          <span
            class="amount ion-text-right"
            slot="end"
          >
             $148.10 
          </span>
        </ion-item>
        <ion-item
          class="failed"
        >
          <ion-label>
            <p
              class="payment-date ion-text-left"
            >
               Aug 10, 2018 
            </p>
            <p
              class="status ion-text-left"
            >
               Failed 
            </p>
            <p
              class="status ion-text-left"
            >
               Credit Card 
            </p>
          </ion-label>
          <span
            class="amount ion-text-right"
            slot="end"
          >
             $148.10 
          </span>
        </ion-item>
        <ion-item
          class="failed"
        >
          <ion-label>
            <p
              class="payment-date ion-text-left"
            >
               Aug 10, 2018 
            </p>
            <p
              class="status ion-text-left"
            >
               Failed 
            </p>
            <p
              class="status ion-text-left"
            >
               Credit Card 
            </p>
          </ion-label>
          <span
            class="amount ion-text-right"
            slot="end"
          >
             $148.10 
          </span>
        </ion-item>
      </ion-list>
    </ion-card-content>
  </ion-card>
</csaa-payment-history-card>
`;
